src/main/java/org/checkerframework/specimin/UnsolvedSymbolVisitor.java

```diff
@@ -877,11 +1089,24 @@ public void updateMissingClass(UnsolvedClass missedClass) {
     while (iterator.hasNext()) {
       UnsolvedClass e = iterator.next();
       if (e.getClassName().equals(missedClass.getClassName())) {
+
+        // add new methods
         for (UnsolvedMethod method : missedClass.getMethods()) {
-          if (!method.getReturnType().equals("")) {
+          boolean alreadyHad = false;
```

Comment:
A brief comment explaining why this boolean is necessary would be helpful. I think the reason is that if the same method is used multiple times in the target(s), it will appear more than once in this loop: once for each call site.